### DISCUSSION TITLE
Do you accept new translations?

### DISCUSSION THREAD

**Discussion by JoseDeFreitas** [alefragnani/vscode-bookmarks#625] [Q&A] [answered]
Hello. I saw the content of the extension is translated into Portuguese (inside the [l10n](https://github.com/alefragnani/vscode-bookmarks/tree/master/l10n) folder) and thought I might contribute translating it into Spanish (which is my native language). After a quick research, however, I found out that a Spanish translation was made a couple of years ago (#427), but it seems the structure of the JSON file has changed. As there's only one language besides English, I wanted to ask first.

**Comment by alefragnani** [comment, marked as answer]
Hi @JoseDeFreitas ,

New translations are welcome, for sure 😬 

About the required resources, it is in fact, composed by three different files, as the translation/localization technologies used by VS Code has evolved over the years. But history aside, you have:

* `./package.nls.xxx.json`: Located in the root folder, these files contains translations for visual elements, like _commands_, _settings_, _menus_ and _walkthroughs_. Basically elements that are contributed in the `package.json` file
* `./l10n/bundle.l10n.xxx.json`: Located inside the `./l10n` folder, these files contains translations for the other resources, like _messages_ that the extension will display during its use.
* `./walkthrough/step.nls.xxx.md`: If the extension provides Walkthroughs, you can translate the step content, when a .md file is available, entirely.

I'm planning to release a new Bookmarks extension this week, providing a walkthrough https://github.com/alefragnani/vscode-bookmarks/pull/621, so I suggest you to wait for this release prior to contribute translation, as you will have new resources to translate.

Thank you

**Reply by JoseDeFreitas** [reply]
Hello, @alefragnani. Thank you for your answer. I see you already merged the pull request #621. May I start translating the files now?

**Reply by alefragnani** [reply]
I think the best option is to wait for the new version to be released, as it will also contains the walkthrough.

This happens because, for now, you can't build the extension yourself (there are some private resources yet), and you will be forced to apply the changes in the installed extension (on `Extensions: Open Extensions Folder`), to test your changes.

Hope this helps

**Reply by JoseDeFreitas** [reply]
Sorry, @alefragnani, I misunderstood your first response and thought the merging of that pull request was the new version being released. I'll start watching the repository to be aware of the new version. Thank you.

**Reply by alefragnani** [reply]
That's ok, no need to be sorry.

On my extensions, a PR being merged won't mean a new release, normally, unless when dealing with hot fixes. I prefer release versions with a pack of features/fixes. Coincidentally, that PR was the last one of the pack.

The 13.4.1 version was released yesterday, so if you update yours, you will see the new walkthrough, and will be able to test your translations.

I hope to find some time to better organize the repo and the building process, so anyone could build and contribute to the extension regularly.

Hope this helps
